modules/nw-ptp-holdover-in-a-grandmaster-clock.adoc

```diff
@@ -0,0 +1,21 @@
+// Module included in the following assemblies:
+//
+// * networking/ptp/configuring-ptp.adoc
+
+:_mod-docs-content-type: CONCEPT
+[id="holdover-in-a-grandmaster-clock_{context}"]
+= Holdover in a grandmaster clock with GNSS as the source
+
+Holdover allows the grandmaster (T-GM) clock to maintain synchronization when the GNSS source is unavailable. During this period, the T-GM clock relies on its internal oscillator and holdover parameters to reduce timing disruptions.
+
+You can define the holdover behavior by configuring the following holdover parameters in the `PTPConfig` custom resource (CR):
+
+* `LocalMaxHoldoverOffSet`: Maximum allowable phase offset during holdover.
+* `LocalHoldoverTimeout`: Maximum time the T-GM clock remains in holdover before transitioning to another state.
+* `MaxInSpecOffset`: Threshold for transitioning from `gm.ClockClass` 6 (`LOCKED`) to `gm.ClockClass` 248 (`FREERUN`). When the phase offset exceeds this threshold, the T-GM clock transitions to the `FREERUN` state. For information on clock class states, see "Grandmaster clock class sync state reference"
+
+The T-GM clock uses the holdover parameters `LocalMaxHoldoverOffSet` and `LocalHoldoverTimeout` to caclulate the slope. Slope is the rate at which the phase offset changes over time. It is measured in picoseconds per second. This value indicates how much the offset increases over a given time period. T-GM clock uses the slope value to predict and compensate for the drift, reducing timing disruptions during holdover.
+
+The T-GM clock uses the following formula to calculate the slope:
+
+* Slope = float(`localMaxHoldoverOffSet`) / float(`localHoldoverTimeout`) * 1000.0
```

Comment:
simplify  this 
slope= localMaxHoldoverOffSet /localHoldoverTimeout ==ns/s 
 PhaseOffset is reported  in ps/s which we convert to ns 
   but when  calculated phase offset is used we convert it to ns/s

Reply:
Hi @aneeshkp 
This part is not clear to me "slope= localMaxHoldoverOffSet /localHoldoverTimeout ==ns/s"
Slope is measured in ps/s but not sure what "==ns/s" means in this context.

Reply:
localMaxHoldoverOffset is measured in nanoseconds, and localHoldoverTimeout is measured in seconds.

Although the DPLL phase offset is in picoseconds, we convert it to nanoseconds. Therefore, the calculated phase offset during holdover is expressed as nanoseconds per second, and the resulting slope is in nanoseconds/secs.

All offsets are consistently declared in nanoseconds in metrics and events .   @vitus133    do you agree with this ?

Reply:
Updated as per the suggestion!
@aneeshkp 

Could you PTAL at the updated content.
Thanks!